spec/sideloading_spec.rb

```diff
@@ -405,6 +413,59 @@ def assert_correct_response
           sl.children[:mastercard].resource_class
         ])
       end
+
+      it 'does not create or require ignored types' do
+        sl = resource.sideload(:credit_card)
+        expect(sl.resource).to be_polymorphic
+        expect(sl.resource.class).to be_abstract_class
+        expect(sl.children.keys).not_to include(:paypal)
+      end
+    end
+
+    context 'when multiple macros defined on the same keys' do
```

Comment:
I do like the test on 417, but would like a more integration-y test like this one as well (actually querying and asserting the json doesn't include the relationship, but without the additional setup needed for this multi-macro use case)

Reply:
I added two more specs; not sure if this is what you were looking for.